nexuse2e-webapp/src/main/frontend/src/app/data/data.service.ts

```diff
@@ -1,13 +1,16 @@
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { environment } from "../../environments/environment";
+import { Message } from "../types";
 
 @Injectable({
   providedIn: "root",
 })
 export class DataService {
   API_URL: string;
 
+  private cache: { [key: string]: unknown } = {};
```

Comment:
Can we add the typeinfo that we having a promise here?
```suggestion
  private cache: { [key: string]: Promise<unknown> } = {};
```